### PR TITLE
Add an `ExperimentRunConfig` object

### LINKED ISSUE #2420: Move ensemble_config, experiment_config, stages_config into one object
This should be in the config module, and validation should be moved from the workspace module along with this new config object. This is also what should be passed around in other modules (such as the engine), instead of passing the triplet of configs.

### PR DESCRIPTION
**Pull request by verveerpj**
**Issue**
Resolves #2420

**Approach**
The `ExperimentRunConfig` just stores the three config objects, and allows retrieving them via properties. On creating of the object, cross validation of the config objects is done.

The choice is made to not make this a `pydantic `object, since we are still keeping three different config objects, that are loaded and validated as `pydantic `objects separately. At this point, the cross-validation code is simple, and the added functionality of `pydantic ` does not seem to be needed.

**Note**
Some validation is still remaining in the workspace object, because it requires knowledge of how resources are implemented in the workspace. That will be fixed in a separate issue (#2419) at which point it should be re-evaluated if that validation should be done elsewhere.

### CODE REVIEW

File: ert3/engine/_run.py [hunk 1]
```diff
@@ -167,7 +173,7 @@ def run(
         .set_type("function" if isinstance(stage, ert3.config.Function) else "unix")
     )
 
-    inputs = ert3.config.link_inputs(ensemble_config, stage)
+    inputs = ert3.config.link_inputs(experiment_run_config.ensemble_config, stage)
```
Comment (jondequinor) [inline ert3/engine/_run.py:176]: Should this be a part of validation in this new configuration object?

The linking was meant to abstract away the fact that inputs are composites of multiple configurations, which is one purpose of the run configuration.

Linked inputs (or just inputs) can then be fetched from the run configuration after linking.

Comment (verveerpj) [reply ert3/engine/_run.py:176]: That sounds like a good idea, I will look into that.

Comment (verveerpj) [reply ert3/engine/_run.py:176]: Done.

File: ert3/config/_experiment_run_config.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    ExperimentConfig,
+    ParametersConfig,
+    SourceNS,
+    StagesConfig,
+    Step,
+)
+
+
+class LinkedInput(NamedTuple):
+    """The LinkedInput class provides all necessary information to us an input
```
Comment (sondreso) [inline ert3/config/_experiment_run_config.py:16]: Spelling mistake at the end of the line, I am at least not able to parse the meaning? :thinking:

**Review by sondreso — APPROVED** [review]
This is very nice :clap: Only a small comment

**Comment by verveerpj** [comment]
Jenkins, please test this

**Pull request merged**